ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -39,9 +39,215 @@
         FDM = ".fdm",
         EMAIL = ".email",
 
-
         Utils = window.CQ.FormsCoreComponents.Utils.v1;
 
+    var JSON_SCHEMA = 'jsonschema',
+        NONE = "none",
+        FORM_DATA_MODEL = "formdatamodel",
+        SCHEMA_REF = "input[name='./schemaRef']",
+        SCHEMA_TYPE = "input[name='./schemaType']",
+        SCHEMA_CONTAINER = ".cmp-adaptiveform-container__schemaselectorcontainer",
+        FDM_CONTAINER = ".cmp-adaptiveform-container__fdmselectorcontainer",
+        SCHEMA_DROPDOWN_SELECTOR = ".cmp-adaptiveform-container__schemaselector",
+        FDM_DROPDOWN_SELECTOR = ".cmp-adaptiveform-container__fdmselector",
+        FORM_MODEL_SELECTOR = ".cmp-adaptiveform-container__selectformmodel",
+        FM_AF_ROOT = "/content/forms/af/",
+        FM_DAM_ROOT ="/content/dam/formsanddocuments/",
+        DAM_SCHEMA_TYPE,
+        DAM_SCHEMA_REF;
+
+    var configuredFormModel,
+        toBeConfiguredFormModel,
+        isConfirmationDialogAccept = false,
+        doNotShowDialogFlag = false,
+        dialogHeader = Granite.I18n.get("Form Model"),
+        dialogContent = "<p>" + Granite.I18n.get("When you replace the old data model with a new model, some data bindings might break and lead to form submission errors.") + "</p>" + "<coral-checkbox id='doNotShowDialogCheckBox'>" + Granite.I18n.get("Don't show this again") + "</coral-checkbox>",
+        dialogFooter = '<button id="formModelDialogCancelButton" is="coral-button" variant="default" coral-close>' + Granite.I18n.get("Cancel") + '</button><button id="formModelDialogAcceptButton" is="coral-button" variant="primary" coral-close>' + Granite.I18n.get("Ok") + '</button>';
+
+    var formModelChangeConfirmationDialog = new Coral.Dialog().set({
+        id: 'formModelChange',
+        header: {
+            innerHTML: dialogHeader
+        },
+        content: {
+            innerHTML: dialogContent
+        },
+        footer: {
+            innerHTML: dialogFooter
+        },
+        variant : 'warning'
+    });
+
+    channel.on("dialog-success", function (e) {
+        configuredFormModel = toBeConfiguredFormModel;
+        isConfirmationDialogAccept = false;
+        if(isForm()){
+            var customEvent = document.createEvent("CustomEvent");
+            customEvent.initCustomEvent("data-model-selected", true, true);
```

Comment:
@im-shiv  Irrespective of any change within dialog or simply opening form container dialog and click on done button will throw "dialog-success" event. And for each such event **data-model-selected** event is being dispatched. This event should be dispatched when there is a real change in data model selection.

Reply:
now, it will only be triggered if data model is changed